walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,66 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
 
 public final class FindMeetingQuery {
+  
+  private Collection<TimeRange> possibleTimes ;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //throw new UnsupportedOperationException("TODO: Implement this method.");
+    // If no attndees
+    if ((request.getAttendees() == null ) && (events == null) ){
+      possibleTimes.add(TimeRange.WHOLE_DAY);
+      return possibleTimes;
+    }
+    //no conflicts
+    else if ((request.getAttendees() != null) && (events == null)) {
+      possibleTimes.add(TimeRange.WHOLE_DAY);
+      return possibleTimes;
+    }
+    //if no events
+    // else if (events == null){
+    //   possibleTimes.add(TimeRange.WHOLE_DAY);
+    //   return possibleTimes;
+    // }
+    
+    int min_duration = (int) request.getDuration();
+    // Initialize min_time to current time
+    int curr_time = TimeRange.START_OF_DAY;
+
+    // Sort array by start time
+    //Collections.sort(events);
+
+    // Loop through all intervals
+    for(Event event: events){
+      int st_tm = event.getWhen().start();
```

Comment:
don't abbreviate variable names unless it's a very common abbreviation (like min), prefer full names as it's clearer.